File: LC-1282/LC-1282.cpp
class Solution {
public:
    /**
     * Concepts: Naive Approach
     */
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        int groupLen = groupSizes.size();
        vector<vector<int>> group, temp(groupLen + 1);
        for (int i = 0; i < groupLen; ++i) {
            temp[groupSizes[i]].push_back(i);
            if (temp[groupSizes[i]].size() == groupSizes[i]) {
                group.push_back(temp[groupSizes[i]]);
                temp[groupSizes[i]].clear();
            }
        }
        return group;
    }
};

File: LC-1282/LC-1282.js
/**
 * @param {number[]} groupSizes
 * @return {number[][]}
 */
var groupThePeople = function (groupSizes) {
  /**
   * Concepts: Naive Approach
   */

  let groupLen = groupSizes.length
  let group = [],
    temp = new Array(groupLen + 1)

  for (let i = 0; i < temp.length; ++i) {
    temp[i] = []
  }

  for (let i = 0; i < groupLen; ++i) {
    temp[groupSizes[i]].push(i)
    if (temp[groupSizes[i]].length === groupSizes[i]) {
      group.push(temp[groupSizes[i]])
      temp[groupSizes[i]] = []
    }
  }
  return group
}
